### PR TITLE
Remove timebounds

### PR DESCRIPTION
**Pull request by hkalodner**
- Remove the GETTIME opcode which was the last place the time bounds were used in the AVM
- Completely remove TimeBounds from ethbridge and validator

### CODE REVIEW

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 57
- Added 10
           

Complexity decreasing per file
==============================
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-validator-core/valprotocol/params.go  -1
+ packages/arb-util/protocol/timebounds.go  -2
+ packages/arb-validator-core/valprotocol/precondition.go  -1
         

Clones removed
==============
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 57
- Added 10
           

Complexity decreasing per file
==============================
+ packages/arb-util/protocol/timebounds.go  -2
+ packages/arb-validator-core/valprotocol/precondition.go  -1
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-validator-core/valprotocol/params.go  -1
         

Clones removed
==============
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 57
- Added 10
           

Complexity decreasing per file
==============================
+ packages/arb-validator-core/valprotocol/params.go  -1
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-util/protocol/timebounds.go  -2
+ packages/arb-validator-core/valprotocol/precondition.go  -1
         

Clones removed
==============
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 57
- Added 10
           

Complexity decreasing per file
==============================
+ packages/arb-validator-core/valprotocol/params.go  -1
+ packages/arb-util/protocol/timebounds.go  -2
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-validator-core/valprotocol/precondition.go  -1
         

Clones removed
==============
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 58
- Added 10
           

Complexity decreasing per file
==============================
+ packages/arb-validator-core/valprotocol/params.go  -1
+ packages/arb-validator-core/valprotocol/precondition.go  -1
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-util/protocol/timebounds.go  -2
         

Clones removed
==============
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 62
- Added 19
           

Complexity decreasing per file
==============================
+ tests/fibgo/connection_test.go  -3
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -2
+ packages/arb-validator-core/valprotocol/params.go  -1
+ packages/arb-util/protocol/timebounds.go  -2
+ packages/arb-validator-core/valprotocol/precondition.go  -1
         

Clones removed
==============
+ tests/fibgo/connection_test.go  -2
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

File: packages/arb-bridge-eth/contracts/vm/ArbFactory.sol [hunk 1]
```diff
@@ -44,7 +44,7 @@ contract ArbFactory is CloneFactory {
         uint128 _gracePeriodTicks,
         uint128 _arbGasSpeedLimitPerTick,
         uint64 _maxExecutionSteps,
-        uint64[2] calldata _maxTimeBoundsWidth,
+        uint64 _maxBlockBoundsWidth,
```
Comment (edfelten) [inline packages/arb-bridge-eth/contracts/vm/ArbFactory.sol:47]: Why do we still need `_maxBlockBoundsWidth`?

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 74
- Added 16
           

Complexity decreasing per file
==============================
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -4
+ tests/fibgo/connection_test.go  -3
+ packages/arb-validator-core/valprotocol/precondition.go  -1
+ packages/arb-validator-core/valprotocol/params.go  -2
         

Clones removed
==============
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ tests/fibgo/connection_test.go  -2
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

File: packages/arb-bridge-eth/contracts/vm/ArbFactory.sol (hunk 1 above)
Comment (hkalodner) [reply packages/arb-bridge-eth/contracts/vm/ArbFactory.sol]: Timebounds have now been completely removed

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 74
- Added 16
           

Complexity decreasing per file
==============================
+ packages/arb-validator/rollup/chainobserver/opinionThread.go  -4
+ packages/arb-validator-core/valprotocol/params.go  -2
+ tests/fibgo/connection_test.go  -3
+ packages/arb-validator-core/valprotocol/precondition.go  -1
         

Clones removed
==============
+ packages/arb-avm-cpp/cmachine/checkpoint_test.go  -1
+ packages/arb-validator/challenges/executionChallenge_test.go  -2
+ packages/arb-compiler-evm/arbitrum/tests/test_eth.py  -13
+ packages/arb-avm-cpp/speedtest/speed_test.go  -2
+ packages/arb-validator/rollup/rollupEthBridge_test.go  -1
+ packages/arb-validator/nodegraph/stakednodegraph_test.go  -1
+ tests/fibgo/connection_test.go  -2
+ packages/arb-provider-go/arbsys.go  -4
+ packages/arb-validator/rollup/chainobserver/chainObserver_test.go  -1
+ packages/arb-validator/nodegraph/nodegraph_test.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5835664&bid=19109708)

**Pull request merged**